### PR TITLE
thunderbird-bin-unwrapped: 115.13.0 → 128.1.0esr

### PR DESCRIPTION
**Pull request by vincentbernat**
## Description of changes

Thunderbird now has an ESR release. Therefore, update to this one (same version as the non-binary version). Also tweak the update script to accept a version prefix and follow ESR this way.

Also see some comments in #307602.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by taku0** [comment]
Running `thunderbird` results in `Segmentation fault`.

File: pkgs/applications/networking/browsers/firefox-bin/update.nix [hunk 1]
```diff
@@ -12,6 +12,7 @@
 , baseName ? "firefox"
 , basePath ? "pkgs/applications/networking/browsers/firefox-bin"
 , baseUrl
+, versionPrefix ? ""
```
Comment (taku0) [inline pkgs/applications/networking/browsers/firefox-bin/update.nix:15]: This should be postfix or suffix.

Comment (vincentbernat) [reply pkgs/applications/networking/browsers/firefox-bin/update.nix]: Fixed.

**Comment by vincentbernat** [comment]
> Running `thunderbird` results in `Segmentation fault`.

It works for me, but I am running it on top of Debian. I don't have yet a desktop with NixOS to test.

File: pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix [hunk 2]
```diff
@@ -203,6 +203,7 @@ stdenv.mkDerivation {
     channel = "release";
     basePath = "pkgs/applications/networking/mailreaders/thunderbird-bin";
     baseUrl = "http://archive.mozilla.org/pub/thunderbird/releases/";
+    versionPrefix = "esr";
```
Comment (taku0) [inline pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix:206]: This also needs to be fixed.

Comment (vincentbernat) [reply pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix]: Sorry. Done.

**Comment by taku0** [comment]
We should rewrite `thunderbird-bin` package to use `autoPatchelfHook` like `firefox-bin` package. Will you rewrite it? If not, I will.

**Comment by vincentbernat** [comment]
Please do, I won't be able to test the result.

**Comment by taku0** [comment]
Added a commit to migrate to `autoPatchelfHook`. Codes are largely borrowed from `firefox-bin`. This works on my NixOS machine.

@lovesegfault, could you review this?

**Comment by vincentbernat** [comment]
Any reason we don't get 128.2.3esr PR automatically? Dunno where to look for a failure.

**Pull request merged**